werickblog/rick_and_morty_api
# FEATURE <id: 169923404>: API rate limit

#### What does this PR do?

- setup API rate limit

#### Description of tasks to be accomplished?

- install relevant packages
- initialize rate limit
- use the initialization as a middleware for the /api/endpoints

#### How can this be manually tested?

- clone the repo
- install relevant packages `yarn install`
- run test `yarn test`, run dev env `yarn dev` & run prod env `yarn start`
- request endpoint `/api/get` 1000 times you should receive a rate limit exceeded error

#### Background Context about this PR?

- advised node version `10+`
- use yarn instead of npm (`npm install -g yarn`)

#### Related Pivotal Tracker story?

[#169923404](https://www.pivotaltracker.com/story/show/169923404)
